src/binding/petsc4py/src/petsc4py/PETSc/DMStag.pyx

```diff
@@ -236,56 +528,260 @@ cdef class DMStag(DM):
+        """
         cdef PetscDMType cval = NULL
         dmtype = str2bytes(dmtype, &cval)
         CHKERR( DMStagSetCoordinateDMType(self.dm, cval) )
 
     # Location slot related functions
 
-    def getLocationSlot(self, loc, c):
+    def getLocationSlot(self, loc: StencilLocation, c: int) -> int:
+        """Return index to use in accessing raw local arrays.
+
+        Not collective.
+
+        Parameters
+        ----------
+        loc
+            Location relative to an element.
+        c
+            Component.
+
+        See Also
+        --------
+        petsc.DMStagGetLocationSlot
+
+        """
         cdef PetscInt slot=0
         cdef PetscInt comp=asInt(c)
         cdef PetscDMStagStencilLocation sloc = asStagStencilLocation(loc)
-        CHKERR( DMStagGetLocationSlot(self.dm, sloc, comp, &slot) ) 
+        CHKERR( DMStagGetLocationSlot(self.dm, sloc, comp, &slot) )
         return toInt(slot)
 
-    def getProductCoordinateLocationSlot(self, loc):
+    def getProductCoordinateLocationSlot(self, loc: StencilLocation) -> None:
+        """Return slot for use with local product coordinate arrays.
+
+        Not collective.
+
+        Parameters
+        ----------
+        loc
+            The grid location.
+
+        See Also
+        --------
+        petsc.DMStagGetProductCoordinateLocationSlot
+
+        """
         cdef PetscInt slot=0
         cdef PetscDMStagStencilLocation sloc = asStagStencilLocation(loc)
         CHKERR( DMStagGetProductCoordinateLocationSlot(self.dm, sloc, &slot) )
         return toInt(slot)
-        
-    def getLocationDof(self, loc):
+
+    def getLocationDof(self, loc: StencilLocation) -> int:
+        """Return number of DOFs associated with a given point on the grid.
+
+        Not collective.
+
+        Parameters
+        ----------
+        loc
+            The grid point.
+
+        See Also
+        --------
+        petsc.DMStagGetLocationDOF
+
+        """
         cdef PetscInt dof=0
         cdef PetscDMStagStencilLocation sloc = asStagStencilLocation(loc)
-        CHKERR( DMStagGetLocationDOF(self.dm, sloc, &dof) ) 
+        CHKERR( DMStagGetLocationDOF(self.dm, sloc, &dof) )
         return toInt(dof)
 
     # Random other functions
 
-    def migrateVec(self, Vec vec, DM dmTo, Vec vecTo):
+    def migrateVec(self, Vec vec, DM dmTo, Vec vecTo) -> None:
+        """Transfer a vector between two ``DMStag`` objects.
+
+        Currently only implemented to migrate global vectors to global vectors.
+
+        Collective.
+
+        Parameters
+        ----------
+        vec
+            The source vector.
+        dmTo
+            The compatible destination object.
+        vecTo
+            The destination vector.
+
+        See Also
+        --------
+        petsc.DMStagMigrateVec
+
+        """
         CHKERR( DMStagMigrateVec(self.dm, vec.vec, dmTo.dm, vecTo.vec ) )
-        
-    def createCompatibleDMStag(self, dofs):
+
+    def createCompatibleDMStag(self, dofs) -> DM:
```

Comment:
Missing annotation. The type could be `Sequence[int]` actually, but for consistency lets use `tuple[int, ...]`
```suggestion
    def createCompatibleDMStag(self, dofs: tuple[int, ...]) -> DM:
```